src/registrar/models/portfolio_invitation.py

```diff
@@ -87,9 +87,10 @@ def retrieve(self):
             raise RuntimeError("Cannot find the user to retrieve this portfolio invitation.")
 
         # and create a role for that user on this portfolio
-        user.portfolio = self.portfolio
+        user_portfolio = user.last_selected_portfolio
+        user_portfolio = self.portfolio
```

Comment:
we do not want the last selected portfolio to change just because they accepted an invite, we want this to add a row in the UserPortfolioPermissions table